addons/point_of_sale/static/src/js/Screens/ClientListScreen/ClientDetailsEdit.js

```diff
@@ -38,25 +22,7 @@ odoo.define('point_of_sale.ClientDetailsEdit', function(require) {
          * Save to field `changes` all input changes from the form fields.
          */
         captureChange(event) {
-            this.changes[event.target.name] = event.target.value;
-        }
-        saveChanges() {
-            let processedChanges = {};
-            for (let [key, value] of Object.entries(this.changes)) {
-                if (this.intFields.includes(key)) {
-                    processedChanges[key] = parseInt(value) || false;
-                } else {
-                    processedChanges[key] = value;
-                }
-            }
-            if ((!this.props.partner.name && !processedChanges.name) ||
-                processedChanges.name === '' ){
-                return this.showPopup('ErrorPopup', {
-                  title: _t('A Customer Name Is Required'),
-                });
-            }
-            processedChanges.id = this.props.partner.id || false;
-            this.trigger('save-changes', { processedChanges });
+            this.props.changes[event.target.name] = event.target.value;
```

Comment:
This is how you introduce complex bug. You never change the value of a props directly like that. Imagine there's a bug linked to an object passed as a props but at a later date, you will not be able to find the source of the change easily because it was changed outside of the object, in a props (that could have any random name), etc.
You can either pass a callback as props if it's coming from the parent component or trigger an event that will be caught by the ancestors components

Reply:
I agree with you in this Jacky. Thanks for pointing it out! 

Also, I have to note that, after the UI simplification task that I completed last week, this PR is now irrelevant. Since the 'Set Customer' button was dropped, the 'workflow' that was causing this issue is no longer accessible. So we could maybe close the PR. What do you think?